### PR TITLE
Moved the majority of the contents of Utils.kt into either InternalUt…

### PR DESCRIPTION
**Pull request by shamsasari**
…ils.kt or KotlinUtils.kt.

What remains is being dealt with in another PR.

### CODE REVIEW

File: core/src/main/kotlin/net/corda/core/Utils.kt [hunk 1]
```diff
@@ -1,49 +1,18 @@
...
-import java.util.zip.Deflater
-import java.util.zip.ZipEntry
-import java.util.zip.ZipInputStream
-import java.util.zip.ZipOutputStream
 
-// TODO: Review by EOY2016 if we ever found these utilities helpful.
-val Int.bd: BigDecimal get() = BigDecimal(this)
-val Double.bd: BigDecimal get() = BigDecimal(this)
-val String.bd: BigDecimal get() = BigDecimal(this)
-val Long.bd: BigDecimal get() = BigDecimal(this)
```
Comment (adagys) [inline core/src/main/kotlin/net/corda/core/Utils.kt]: are we removing these altogether?

Comment (shamsasari) [inline core/src/main/kotlin/net/corda/core/Utils.kt]: Yep, they weren't used as far as I can remember and they're not very clear

**Comment by adagys** [comment]
also what's the effect of moving stuff into core.internal, is the usage restricted in any way, or is it just to indicate it shouldn't be used

**Comment by shamsasari** [comment]
Right now we're just saying that it's not part of our API and must not be used (which reminds me I need to update the docs to say so somewhere). At some point we'll put restrictions on CorDapps where they won't load if they use internal API.

**Comment by joeldudleyr3** [comment]
LGTM too

**Pull request merged**